shared/src/question-generators/math/modFactorization.ts

```diff
@@ -0,0 +1,217 @@
+  id: "modfactor",
+  name: tFunctional(translations, "name"),
+  description: tFunctional(translations, "description"),
+  tags: ["factorization", "modulus", "arithmetic"],
+  languages: ["en", "de"],
+  author: "Janette Welker",
+  license: "MIT",
+  expectedParameters: [
+    {
+      type: "string",
+      name: "variant",
+      allowedValues: ["large_number", "large_exponent", "multiplication"],
+    },
+  ],
+
+  generate: (lang, parameters, seed) => {
+    const path = serializeGeneratorCall({
+      generator: modFactor,
+      lang,
+      parameters,
+      seed,
+    })
+
+    const random = new Random(seed)
+    const variant = parameters.variant as "large_number" | "large_exponent" | "multiplication"
+
+    switch (variant) {
+      case "large_number":
+        return generateFactorLargeNumberQuestion(lang, path, random)
+      case "multiplication":
+        return generateFactorMultiplicationQuestion(lang, path, random)
+      case "large_exponent":
+        return generateFactorLargeExponentQuestion(lang, path, random)
+      default:
+        throw new Error("Unknown variant")
+    }
+  },
+}
+
+// Large number
+function generateFactorLargeNumberQuestion(lang: Language, path: string, random: Random) {
+  const n = random.int(2, 20)
+  const factors = generateFactors(random, 2, 3, 10, 50)
+  const x = factors.reduce((acc, factor) => acc * factor, 1)
+  const calculation =
+    factors.map((factor) => `(${factor} \\pmod{${n}})`).join(" \\times ") + ` \\equiv ${x % n}`
+
+  const question: FreeTextQuestion = {
+    type: "FreeTextQuestion",
+    name: modFactor.name(lang),
+    path: path,
+    text: t(translations, lang, "factorLargeNumberQuestion", { x: String(x), n: String(n) }),
+    prompt: `$x \\equiv $`,
+    bottomText: t(translations, lang, "bottomText"),
+    feedback: generateModularFeedback(lang, x, n, calculation),
+  }
+
+  return { question, testing: { x, n, calculation } }
+}
+
+// Modular multiplication
+function generateFactorMultiplicationQuestion(lang: Language, path: string, random: Random) {
+  // sample desired result between (0,n-1) for less repetitive solutions
+  const n = random.int(2, 20)
+  const result = random.int(0, n - 1)
+
+  let a, b
+  let factorsA, factorsB
+
+  // try generating factors until desired result is achieved
+  let attempts = 0
+  const maxAttempts = 1000
+  let allowOneAsFactor = false
+
+  do {
+    factorsA = generateFactors(random, 2, 3, allowOneAsFactor ? 1 : 2, 50)
+    factorsB = generateFactors(random, 2, 3, allowOneAsFactor ? 1 : 2, 50)
+    a = factorsA.reduce((acc, factor) => acc * factor, 1)
+    b = factorsB.reduce((acc, factor) => acc * factor, 1)
+    attempts++
+
+    if (attempts >= maxAttempts) {
+      allowOneAsFactor = true
+    }
+  } while ((a * b) % n !== result && attempts < maxAttempts + 1000)
+  if ((a * b) % n !== result) {
+    throw new Error(`Timeout at factor generation after ${maxAttempts + 1000} attempts`)
```

Comment:
How likely is this error? Or instead of throwing an error, can we maybe change the calculation such that we always generate a question. 
But if the error is very unlikely just ignore :-)